_analyzers/token-filters/delimited-term-frequency.md

```diff
@@ -0,0 +1,276 @@
+        {
+          "token": "foo|100",
+          "start_offset": 0,
+          "end_offset": 7,
+          "type": "word",
+          "position": 0,
+          "termFrequency": 1
+        }
+      ]
+    },
+    "tokenfilters": [
+      {
+        "name": "delimited_term_freq",
+        "tokens": [
+          {
+            "token": "foo",
+            "start_offset": 0,
+            "end_offset": 7,
+            "type": "word",
+            "position": 0,
+            "termFrequency": 100
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+## Custom `delimited_term_freq` token filter
+
+To configure a custom `delimited_term_freq` token filter, first specify the delimiter in the mapping request, in this example, `^`:
+
+```json
+PUT /testindex
+{
+  "settings": {
+    "analysis": {
+      "filter": {
+        "my_delimited_term_freq": {
+          "type": "delimited_term_freq",
+          "delimiter": "^"
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Then analyze text with the custom token filter you created:
+
+```json
+POST /testindex/_analyze
+{
+  "text": "foo^3",
+  "tokenizer": "keyword",
+  "filter": ["my_delimited_term_freq"],
+  "attributes": ["termFrequency"],
+  "explain": true
+}
+```
+{% include copy-curl.html %}
+
+The response contains both the original token and the parsed version with the term frequency:
+
+```json
+{
+  "detail": {
+    "custom_analyzer": true,
+    "charfilters": [],
+    "tokenizer": {
+      "name": "keyword",
+      "tokens": [
+        {
+          "token": "foo|100",
+          "start_offset": 0,
+          "end_offset": 7,
+          "type": "word",
+          "position": 0,
+          "termFrequency": 1
+        }
+      ]
+    },
+    "tokenfilters": [
+      {
+        "name": "delimited_term_freq",
+        "tokens": [
+          {
+            "token": "foo",
+            "start_offset": 0,
+            "end_offset": 7,
+            "type": "word",
+            "position": 0,
+            "termFrequency": 100
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+## Combining `delimited_token_filter` with scripts
+
+You can write Painless scripts to calculate custom scores for the documents in the results.
+
+First, create an index and provide the following mappings and settings:
+
+```json
+PUT /test
+{
+  "settings": {
+    "number_of_shards": 1,
+    "analysis": {
+      "tokenizer": {
+        "keyword_tokenizer": {
+          "type": "keyword"
+        }
+      },
+      "filter": {
+        "my_delimited_term_freq": {
+          "type": "delimited_term_freq",
+          "delimiter": "^"
+        }
+      },
+      "analyzer": {
+        "custom_delimited_analyzer": {
+          "tokenizer": "keyword_tokenizer",
+          "filter": ["my_delimited_term_freq"]
+        }
+      }
+    }
+  },
+  "mappings": {
+    "properties": {
+      "f1": {
+        "type": "keyword"
+      },
+      "f2": {
+        "type": "text",
+        "similarity": "BM25",
```

Comment:
My bad, this `similarity` setup is not closely related within the context of this example. I am fine with removing it.